internal/worker/pubsub/subscriber.go

```diff
@@ -239,34 +245,37 @@ func (s *subscriber) apiServerChanges(topic string, details apiserver.Details, e
 	}
 	for name, server := range s.servers {
 		if !apiServers.Contains(name) {
-			s.config.Logger.Debugf(context.TODO(), "%s no longer listed as an apiserver", name)
+			s.config.Logger.Debugf(ctx, "%s no longer listed as an apiserver", name)
 			server.Kill()
 			err := server.Wait()
 			if err != nil {
-				s.config.Logger.Errorf(context.TODO(), "%v", err)
+				s.config.Logger.Errorf(ctx, "%v", err)
 			}
 			delete(s.servers, name)
 		}
 	}
-	s.config.Logger.Tracef(context.TODO(), "update complete")
+	s.config.Logger.Tracef(ctx, "update complete")
 }
 
 func (s *subscriber) forwardMessage(topic string, data map[string]interface{}) {
+	ctx, cancel := s.scopedContext()
+	defer cancel()
```

Comment:
Should this not be passed in?

Reply:
No, the pubsub handlers/callbacks doesn't take a context, you have to spawn a new one.